File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> mySet = new HashSet<>();
        for(int num: nums){
            mySet.add(num);
        }

        int seqLen = 0;
        for(int i : mySet){
            
            if(!(mySet.contains(i-1))){
                int currNum = i;
                int currentStreak = 1;
                while(mySet.contains(currNum+1)){
                    currNum++;
                    currentStreak++;
                }
                seqLen = Math.max(currentStreak, seqLen);
            }
            


        }
        return seqLen;
        
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        set_nums = set(nums)
        longest = 0

        for i in set_nums:
            if (i - 1) not in set_nums:
                length = 1
                while ((i+length) in set_nums):
                    length += 1
                longest = max(length, longest)

        return longest 



        